From dba0e5f43047f1bb1f4c9af39135b2af27c6e3ac Mon Sep 17 00:00:00 2001
From: Bartosz Golaszewski <bgolaszewski@baylibre.com>
Date: Mon, 1 Aug 2016 18:04:25 +0200
Subject: [PATCH 13/16] dt: acme: add gpio multiplexer for i2c1

Use the i2c-mux-gpio module to automatically control the two i2c lines
behind a gpio-driven mux on i2c bus 1.

Signed-off-by: Bartosz Golaszewski <bgolaszewski@baylibre.com>
---
 arch/arm/boot/dts/am335x-boneblack.dts | 67 ++++++++++++++++++++++++++++++++++
 1 file changed, 67 insertions(+)

diff --git a/arch/arm/boot/dts/am335x-boneblack.dts b/arch/arm/boot/dts/am335x-boneblack.dts
index 55c0e95..1ace977 100644
--- a/arch/arm/boot/dts/am335x-boneblack.dts
+++ b/arch/arm/boot/dts/am335x-boneblack.dts
@@ -13,6 +13,37 @@
 / {
 	model = "TI AM335x BeagleBone Black";
 	compatible = "ti,am335x-bone-black", "ti,am335x-bone", "ti,am33xx";
+
+	i2cmux@1 {
+		compatible = "i2c-mux-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		mux-gpios = <&pca9534 0 GPIO_ACTIVE_LOW>;
+		i2c-parent = <&i2c1>;
+		idle-state = <1>;
+
+		i2c@1 {
+			reg = <1>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			/* cape eeproms go here */
+		};
+
+		i2c@2 {
+			reg = <2>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			pca9535: pca9535@20 {
+				compatible = "pca9535";
+				reg = <0x20>;
+				#gpio-cells = <2>;
+			};
+
+			/* ina231 definitions go here */
+		};
+	};
 };
 
 &ldo3_reg {
@@ -64,6 +95,20 @@
 			AM33XX_IOPAD(0x9b0, PIN_OUTPUT_PULLDOWN | MUX_MODE3)	/* xdma_event_intr0 */
 		>;
 	};
+
+	i2c1_pins: pinmux_i2c1_pins {
+		pinctrl-single,pins = <
+			0x158 (PIN_INPUT_PULLUP | MUX_MODE2)
+			0x15c (PIN_INPUT_PULLUP | MUX_MODE2)
+		>;
+	};
+
+	i2c2_pins: pinmux_i2c2_pins {
+		pinctrl-single,pins = <
+			0x178 (PIN_INPUT_PULLUP | MUX_MODE3)
+			0x17c (PIN_INPUT_PULLUP | MUX_MODE3)
+		>;
+	};
 };
 
 &lcdc {
@@ -91,6 +136,28 @@
 	};
 };
 
+&i2c1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c1_pins>;
+
+	status = "okay";
+	clock-frequency = <400000>;
+
+	pca9534: pca9534@21 {
+		compatible = "pca9534";
+		reg = <0x21>;
+		#gpio-cells = <2>;
+	};
+};
+
+&i2c2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&i2c2_pins>;
+
+	status = "okay";
+	clock-frequency = <400000>;
+};
+
 &rtc {
 	system-power-controller;
 };
-- 
2.7.0

